client/p2p.go

```diff
@@ -12,9 +12,17 @@ package client
 
 import (
 	"context"
+
+	"github.com/libp2p/go-libp2p/core/peer"
 )
 
+// P2P is a peer connected database implementation.
 type P2P interface {
+	DB
```

Comment:
todo: It took me a while to figure it out, but you have removed functionality here.  `client.P2P` used to respect the transactionality of `client.Store`, but now it only works with implicit transactions.

I'm not sure we wish to lose this, and this change should be made very visible and open to discussion with the team.

Reply:
I discussed this with @fredcarle and forgot to make a note. We came to the conclusion that peer operations only make sense when done implicitly. It would be pretty difficult to have the transaction also rollback the peer connections that happen during these operations.

Reply:
We are not losing functionality. We are removing the need for the `Store` interface to support P2P related actions. The functionality still exists but is handled all within the `net` package. However, whatever implements the `P2P` interface needs access to a db instance and thus support the `DB` interface. Action related to the `P2P` interface can still use transactions.

Reply:
> We came to the conclusion that peer operations only make sense when done implicitly.

I do disagree with this, with the obvious example being during an update to a new application version - there may be several of these operations (plus other stuff, like schema updates etc) and if any one of those operations fails, they should all fail so that the update is not partially applied.

I will not push back against the both of you on this, but I do think explicit transactions here are useful.